addons/web/static/src/core/registry_hook.js

```diff
@@ -0,0 +1,28 @@
+import { useState } from "@odoo/owl";
+import { useBus } from "@web/core/utils/hooks";
+
+/**
+ * Get the entries of a registry and update the state when the registry is updated.
+ * @param {import('./registry').Registry} registry
+ * @returns {{ entries: [string, any][] }} the entries of the registry, as a reactive state
+ */
+export function useRegistry(registry) {
+    const state = useState({ entries: registry.getEntries() });
+    useBus(registry, "UPDATE", ({ detail }) => {
+        const index = state.entries.findIndex(([k]) => k === detail.key);
+        if (detail.operation === "add" && index === -1) {
+            // push the new entry at the right place
+            const newEntries = registry.getEntries();
+            const newEntry = newEntries.find(([k]) => k === detail.key);
+            const newIndex = newEntries.indexOf(newEntry);
+            if (newIndex === newEntries.length - 1) {
+                state.entries.push(newEntry);
+            } else {
+                state.entries.splice(newIndex, 0, newEntry);
+            }
+        } else if (detail.operation === "delete" && index >= 0) {
+            state.entries.splice(index, 1);
+        }
+    });
```

Comment:
if there's an order issue, shouldn't it be fixed in the registry directly?

Reply:
Do you mean that in the faulty test, the operation on the registry isn't 'delete' nor 'add'? So this is only about filtering out some operations? I agree to address this post merge